sympy/tensor/functions.py

```diff
@@ -61,3 +62,116 @@ def __getitem__(self, index):
             arg.__getitem__(tuple(next(index) for i in shp))
             for arg, shp in zip(self.args, self._get_args_shapes())
         )
+
+
+def shape(expr):
+    """
+    Return the shape of the array as a tuple.
```

Comment:
Matrices are a more common use in sympy than arrays so the word matrix should be here in the first line of this function. The vast majority of the time this will be used to get the shape of a matrix and not an array.

I know that you have talked about unifying matrices and arrays but they are not unified now. We should write docs now that correctly specify the current behaviour. (SymPy already has docs written based on hopeful future changes to the codebase and they are highly problematic)

Reply:
Maybe this docstring should have a parameters section and a raises section that references NoShapeError. The NoShapeError docstring should show how to import and catch that error.